src/aws-elasticache/user.ts

```diff
@@ -0,0 +1,336 @@
+import {
+  IResource,
+  Lazy,
+  Names,
+  Resource,
+  Stack,
+  aws_elasticache,
+  SecretValue,
+  Token,
+  aws_iam,
+  ArnFormat,
+} from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { Engine } from './util';
+
+/**
+ * An User
+ */
+export interface IUser extends IResource {
+  /**
+   * The ARN of the user.
+   *
+   * @attribute
+   */
+  readonly userArn: string;
+  /**
+   * The ID of the user.
+   *
+   * @attribute
+   */
+  readonly userId: string;
+
+  /**
+   * Grant the given identity the specified actions
+   */
+  grant(grantee: aws_iam.IGrantable, ...actions: string[]): aws_iam.Grant;
+
+  /**
+   * Grant the given identity connection access to the cache.
+   */
+  grantConnect(grantee: aws_iam.IGrantable): aws_iam.Grant;
+}
+
+/**
+ * Authentication type.
+ */
+export enum AuthenticationType {
+  /**
+   * Password required.
+   */
+  PASSWORD = 'password',
+
+  /**
+   * No password required.
+   */
+  NO_PASSWORD_REQUIRED = 'no-password-required',
+
+  /**
+   * IAM authentication.
+   */
+  IAM = 'iam',
+}
+
+/**
+ * Properties for defining an User.
+ */
+export interface UserProps {
+  /**
+   * The username of the user.
+   *
+   * The name can have up to 120 characters, and must not contain spaces.
+   *
+   * \`userId\` and \`userName\` must be same When \`authenticationType\` is set to \`AuthenticationType.IAM\`.
+   *
+   * @default - same value as `userId`
+   * @see https://docs.aws.amazon.com/AmazonElastiCache/latest/dg/auth-iam.html
+   */
+  readonly userName?: string;
+
+  /**
+   * The ID of the user.
+   *
+   * \`userGroupId\` can have up to 40 characters.
+   *
+   * \`userId\` must consist only of alphanumeric characters or hyphens, with the first character as a letter,
+   * and it can't end with a hyphen or contain two consecutive hyphens.
+   *
+   * \`userId\` and \`userName\` must be same When \`authenticationType\` is set to \`AuthenticationType.IAM\`.
+   *
+   * @default - auto generate
+   */
+  readonly userId?: string;
+
+  /**
+   * Access permissions string used for this user.
+   *
+   * @default - 'off -@all'
```

Comment:
How about setting it to `on ~* +@all` by default? There can be tradeoffs between security and usability, but I would prefer that the default values just work.

Reply:
I also like this suggestion, but I will wait for the maintainer's opinion.